tests/test_compat.py

```diff
@@ -73,123 +74,132 @@ async def test_running_tasks(self):
+    async def test_lock_release(self) -> None:
         lock = Lock()
         lock.acquire_nowait()
         with pytest.deprecated_call():
             await lock.release()
 
-    async def test_memory_object_stream_send_nowait(self):
+    async def test_memory_object_stream_send_nowait(self) -> None:
         send, receive = create_memory_object_stream(1)
         with pytest.deprecated_call():
             await send.send_nowait(None)
 
-    async def test_semaphore_release(self):
+    async def test_semaphore_release(self) -> None:
         semaphore = Semaphore(1)
         semaphore.acquire_nowait()
         with pytest.deprecated_call():
             await semaphore.release()
 
-    async def test_move_on_after(self):
+    async def test_move_on_after(self) -> None:
         with pytest.deprecated_call():
             async with move_on_after(0):
                 pass
 
-    async def test_fail_after(self):
+    async def test_fail_after(self) -> None:
         with pytest.raises(TimeoutError), pytest.deprecated_call():
             async with fail_after(0):
                 pass
 
-    async def test_run_sync_in_worker_thread(self):
+    async def test_run_sync_in_worker_thread(self) -> None:
         with pytest.deprecated_call():
             thread_id = await run_sync_in_worker_thread(threading.get_ident)
             assert thread_id != threading.get_ident()
 
-    async def test_run_async_from_thread(self):
-        async def get_ident():
+    async def test_run_async_from_thread(self) -> None:
+        async def get_ident() -> int:
             return threading.get_ident()
 
-        def thread_func():
+        def thread_func() -> int:
             with pytest.deprecated_call():
                 return run_async_from_thread(get_ident)
 
         assert await to_thread.run_sync(thread_func) == threading.get_ident()
 
-    async def test_run_sync_from_thread(self):
-        def thread_func():
+    async def test_run_sync_from_thread(self) -> None:
+        def thread_func() -> int:
             with pytest.deprecated_call():
                 return run_sync_from_thread(threading.get_ident)
 
         assert await to_thread.run_sync(thread_func) == threading.get_ident()
 
-    async def test_current_default_worker_thread_limiter(self):
+    async def test_current_default_worker_thread_limiter(self) -> None:
         with pytest.deprecated_call():
             default_limiter = to_thread.current_default_thread_limiter()
             assert current_default_worker_thread_limiter() is default_limiter
 
-    async def test_create_blocking_portal(self):
+    async def test_create_blocking_portal(self) -> None:
         with pytest.deprecated_call():
             async with create_blocking_portal():
                 pass
 
 
 class TestPickle:
-    def test_deprecated_awaitable_none(self):
-        obj = DeprecatedAwaitable(threading.get_ident)
+    def test_deprecated_awaitable_none(self) -> None:
+        def fn() -> DeprecatedAwaitable:
```

Comment:
* couple of fixes here, `DeprecatedAwaitable` takes a `Callable[..., DeprecatedAwaitable]` and `threading.get_ident` returns an `int` so is incompatible.
* also threading.get_ident is pickle-able so won't trigger an error here if it's accidentally included